src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php

```diff
@@ -221,6 +221,10 @@ public function prepare_item_for_response( $item, $request ) {
 			$data['slug'] = $post_type->name;
 		}
 
+		if ( in_array( 'icon', $fields, true ) ) {
```

Comment:
```suggestion
		if ( rest_is_field_included( 'icon', $fields ) ) {
```

Reply:
Should this be updated to all the other checks? If yes, it would be better to have a separate PR, no?

Reply:
I don't mind doing this in this PR.